Packs/ADPITAdmin/Integrations/ADP_IT_Admin/ADP_IT_Admin.py

```diff
@@ -0,0 +1,202 @@
+    def get_workers_async(self):
+        uri_suffix = '/hr/v2/workers'
+        query_params = {
+            'count': 'true'
+        }
+        headers = self._headers
+        headers['prefer'] = 'respond-async'
+
+        return self._http_request(method="GET", url_suffix=uri_suffix, headers=headers, resp_type='response',
+                                  params=query_params, cert=self.cert_file)
+
+    def get_workers_by_uri(self, uri):
+        return self.get_paged_workers(uri)
+
+    def get_paged_workers(self, uri):
+        workers_list = []
+        res = self._http_request(method="GET", url_suffix=uri, cert=self.cert_file, resp_type='response')
+        if res.status_code == 200:
+            workers = res.json().get('workers', [])
+            workers_list.extend(workers)
+
+        while "location" in res.headers and res.status_code == 200:
+            next_page_uri = res.headers.get('location')
+            res = self._http_request(method="GET", url_suffix=next_page_uri, cert=self.cert_file, resp_type='response')
+            workers = res.json().get('workers', [])
+            workers_list.extend(workers)
+        return workers_list
+
+    def get_workers_by_associate_oid(self, associate_oid):
+        uri_suffix = f'/hr/v2/workers/{associate_oid}'
+        return self._http_request(method="GET", url_suffix=uri_suffix, cert=self.cert_file)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client, args):
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    Args:
+        client: HelloWorld client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+
+    uri_suffix = '/hr/v2/workers'
+    query_params = {
+        '$top': '1'
+    }
+
+    client._http_request(method="GET", url_suffix=uri_suffix, params=query_params, cert=client.cert_file)
+    return 'ok', None, None
+
+
+def get_worker_command(client, args):
+    res = client.get_workers_by_associate_oid(args.get('associateOID'))
+    readable_output = tableToMarkdown('ADP Worker:', res)
+
+    return (
+        readable_output,
+        {},
+        res
+    )
+
+
+def get_all_workers_trigger_async_command(client, args):
+    res = client.get_workers_async()
+    retry_after = res.headers["Retry-After"]
+    # Get next URI from the response headers
+    async_uri = res.links.get('/adp/processing-status', {}).get('url')
+
+    outputs = {
+        'ADP': {
+            'WorkersURI': async_uri,
+            "RetryAfter": retry_after
+        }
+    }
+
+    readable_output = tableToMarkdown('ADP Async Response:', outputs.get('ADP'))
+
+    return (
+        readable_output,
+        outputs,
+        None
+    )
+
+
+def get_all_workers_command(client, args):
+
+    workers_uri = args.get('workersURI')
+    workers = client.get_workers_by_uri(workers_uri)
+
+    return (
+        None,
```

Comment:
See my note about this command's output in the yml file. Either if we retrieve data or not, let's give any indication that the execution succeeded (and add it to the readme as well).

Reply:
Done